int32 sharpness
---
int32 error